File: frontend/src/types/tenant.ts
/**
 * Tenant Management Types
 * Defines the structure for tenants (business entities) in the multi-tenant system
 */

import type { Address } from './address';

export type TenantStatus = 'active' | 'inactive';

export interface TenantSettings {
  maxMarkets?: number;    // Limit on number of markets
  maxUsers?: number;      // Limit on number of users
  features?: string[];    // Enabled feature flags
}

export interface Tenant {
  id: string;
  name: string;           // Unique identifier (slug)
  displayName: string;    // Human-readable name
  status: TenantStatus;
  contactEmail: string;
  contactPhone?: string;
  address?: Address;
  settings?: TenantSettings;
  marketCount?: number;   // Number of associated markets
  createdAt: Date;
  updatedAt: Date;
  createdBy?: string;     // User ID
  updatedBy?: string;     // User ID
}

export interface CreateTenantInput {
  name: string;
  displayName: string;
  contactEmail: string;
  contactPhone?: string;
  address?: Address;
  settings?: TenantSettings;
}

export interface UpdateTenantInput {
  displayName?: string;
  status?: TenantStatus;
  contactEmail?: string;
  contactPhone?: string;
  address?: Address;
  settings?: TenantSettings;
}

export interface TenantFilters {
  search?: string;
  status?: TenantStatus;
  page?: number;
  limit?: number;
}

export interface TenantsResponse {
  data: Tenant[];
  total: number;
  page: number;
  limit: number;
}

File: api/EComm.Data/Entities/Tenant.cs
using EComm.Data.ValueObjects.Common;
using EComm.Data.ValueObjects.Tenant;

namespace EComm.Data.Entities;

public class Tenant
{
    public string Id { get; set; } = string.Empty;
    public string Name { get; set; } = string.Empty;
    public string DisplayName { get; set; } = string.Empty;
    public string Status { get; set; } = "active";
    public string ContactEmail { get; set; } = string.Empty;
    public string? ContactPhone { get; set; }
    public Address? Address { get; set; }
    public TenantSettings? Settings { get; set; }
    public int MarketCount { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
}
